### PR TITLE
[REVIEW] Update Java bindings to specify timestamp units for ORC and Parquet readers

### PR DESCRIPTION
**Pull request by nartal1**
Update Java API to be in sync with libcudf.
https://github.com/rapidsai/cudf/pull/2585 was merged recently.

### CODE REVIEW

File: java/src/main/java/ai/rapids/cudf/Table.java [hunk 1]
```diff
@@ -183,9 +183,10 @@ public String toString() {
    * @param filePath          the path of the file to read, or null if no path should be read.
    * @param address           the address of the buffer to read from or 0 if we should not.
    * @param length            the length of the buffer to read from.
+   * @param timeUnit              return type of TimeStamp in units
```
Comment (abellina) [inline java/src/main/java/ai/rapids/cudf/Table.java]: nit, spacing a bit off here.

Comment (nartal1) [reply java/src/main/java/ai/rapids/cudf/Table.java]: @abellina Thanks for noticing this :) PTAL

File: java/src/test/java/ai/rapids/cudf/TableTest.java [hunk 2]
```diff
@@ -595,10 +595,10 @@ void testReadORCFull() {
   @Test
   void testReadORCNumPyTypes() {
     assumeTrue(Cuda.isEnvCompatibleForTesting());
-    // by default ORC will promote date and timestamp columns to DATE64
+    // by default ORC will promote DATE32 to DATE64
```
Comment (abellina) [inline java/src/test/java/ai/rapids/cudf/TableTest.java]: nit, I feel this comment could say something about `TIMESTAMP`. Is it that now `DATE32` is promoted to `TIMESTAMP`?

File: java/src/test/java/ai/rapids/cudf/TableTest.java [hunk 3]
```diff
@@ -611,6 +611,20 @@ void testReadORCNumPyTypes() {
     }
   }
 
+  @Test
+  void testReadORCTimeUnit() {
+    assumeTrue(Cuda.isEnvCompatibleForTesting());
+    // specifying no NumPy types should load them as DATE32 and TIMESTAMP
+    // specifying TimeUnit will return the result in that unit
+    ORCOptions opts = ORCOptions.builder().withNumPyTypes(false).withTimeUnit(TimeUnit.SECONDS).build();
```
Comment (abellina) [inline java/src/test/java/ai/rapids/cudf/TableTest.java]:
```suggestion
    ORCOptions opts = ORCOptions.builder()
      .withNumPyTypes(false)
      .withTimeUnit(TimeUnit.SECONDS).build();
```

Comment (abellina) [reply java/src/test/java/ai/rapids/cudf/TableTest.java]: Just breaking it so it's not such a long line.

**Comment by abellina** [comment]
@nartal1 I had a few minor comments.

**Comment by nartal1** [comment]
Thanks @abellina . Could you PTAL.

**Comment by nartal1** [comment]
Thanks @abellina  and @kuhushukla .

**Pull request merged**